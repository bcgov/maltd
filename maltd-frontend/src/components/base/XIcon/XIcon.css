.inline-block {
  display: inline-block !important;
  margin-top: -10px;
  margin-left: 20px;
}
