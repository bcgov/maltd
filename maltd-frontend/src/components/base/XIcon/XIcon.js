import React from "react";
import PropTypes from "prop-types";
import Icon from "../../../img/x-icon.PNG";
import "./XIcon.css";

export default function XIcon({ onClick, id }) {
  return (
    <div
      id="main"
      role="button"
      tabIndex="0"
      onClick={() => onClick(id)}
      onKeyDown={() => onClick(id)}
      className="inline-block"
    >
      <img src={Icon} alt="x icon" width="25px" height="25px" />
    </div>
  );
}

XIcon.propTypes = {
  onClick: PropTypes.func.isRequired,
  id: PropTypes.string
};
