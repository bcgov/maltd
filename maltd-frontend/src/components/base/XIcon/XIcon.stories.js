import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import "bootstrap/dist/css/bootstrap.css";
import XIcon from "./XIcon";

storiesOf("XIcon", module).add("default", () => (
  <XIcon id="12" onClick={action("x icon clicked")} />
));
