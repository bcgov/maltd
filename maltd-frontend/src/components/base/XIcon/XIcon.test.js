import React from "react";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { shallow } from "enzyme";
import XIcon from "./XIcon";

Enzyme.configure({ adapter: new Adapter() });

describe("X Icon", () => {
  test("Component renders as expected", () => {
    const component = renderer.create(
      <XIcon id="123" onClick={() => jest.fn()} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Component onClick and onKeyDown work as expected", () => {
    const testState = { isClicked: false };
    const onClick = () => {
      testState.isClicked = true;
    };

    const wrapper = shallow(
      <XIcon id="123" onKeyDown={onClick} onClick={onClick} />
    );

    wrapper.find("#main").simulate("keydown");
    expect(testState.isClicked).toEqual(true);

    wrapper.find("#main").simulate("click");
    expect(testState.isClicked).toEqual(true);
  });
});
