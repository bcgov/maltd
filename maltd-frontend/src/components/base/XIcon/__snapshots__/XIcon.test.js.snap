// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`X Icon Component renders as expected 1`] = `
<div
  className="icon"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <img
    alt="x icon"
    height="25px"
    src={
      Object {
        "process": [Function],
      }
    }
    width="25px"
  />
</div>
`;
